### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Changes proposed in this pull request:
Spelling fixes
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

http://www.softether.org/5-download/src/9.patch
Option 1 from the above policy.

### CODE REVIEW

**Comment by jl91569** [comment]
I noticed you changed "Flagment" to "Gragment", I'm pretty sure it's meant to be "Fragment".

`UCHAR FlagsAndGragmentOffset[2]; // Flag and Fragment offset`

**Comment by jsoref** [comment]
Oops. I probably hand typed that correction (from memory it took me a long time to figure out what `Flagment` was because of `FlagsAndFragmentOffset` -- although the comment in the header did give away the answer `// The flag and fragment offset`) and my finger slipped... Fixed. Thanks.

File: src/Cedar/SM.c [hunk 1]
```diff
@@ -13776,7 +13776,7 @@ void SmEditUserDlgOk(HWND hWnd, SM_EDIT_USER *s)
 		}
 		FreeRpcSetUser(&t);
 
-		MsgBoxEx(hWnd, MB_ICONINFORMATION, _UU("SM_USER_CREEATE_OK"), u->Name);
+		MsgBoxEx(hWnd, MB_ICONINFORMATION, _UU("SM_USER_CREATE_OK"), u->Name);
```
Comment (moatazelmasry2) [inline src/Cedar/SM.c]: Add to Korean

File: src/Cedar/SW.c [hunk 2]
```diff
@@ -4155,7 +4155,7 @@ void SwInstallShortcuts(SW *sw, WIZARD_PAGE *wp, SW_COMPONENT *c, SW_TASK *t)
 		{
 			// Show the error message if it fails
 			UINT msgret;
-			UniFormat(tmp, sizeof(tmp), _UU("SW_PERFORM_MSG_CRAETE_LINK_ERROR"), lnk_fullpath);
+			UniFormat(tmp, sizeof(tmp), _UU("SW_PERFORM_MSG_CREATE_LINK_ERROR"), lnk_fullpath);
```
Comment (moatazelmasry2) [inline src/Cedar/SW.c]: Fix in Korean

File: src/Cedar/WinUi.c [hunk 3]
```diff
@@ -246,7 +246,7 @@ UINT UpdateConfigDlgProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam, void
 LABEL_CLOSE:
 		LoadUpdateUiSetting(u, &s);
 
-		s.DisableCheck = IsChecked(hWnd, S_DISBLE);
+		s.DisableCheck = IsChecked(hWnd, S_DISABLE);
```
Comment (moatazelmasry2) [inline src/Cedar/WinUi.c]: Fix in Korean

File: src/Mayaqua/Table.c [hunk 4]
```diff
@@ -1440,7 +1440,7 @@ bool LoadTableMain(wchar_t *filename)
 
 	FreeBuf(b);
 
-	SetLocale(_UU("DEFAULE_LOCALE"));
+	SetLocale(_UU("DEFAULT_LOCALE"));
```
Comment (moatazelmasry2) [inline src/Mayaqua/Table.c]: Fix in Korean

File: src/PenCore/PenCore.rc [hunk 5]
```diff
@@ -2950,7 +2950,7 @@ BEGIN
     GROUPBOX        "@STATIC1",-1,7,5,265,51
     ICON            ICO_NIC_ONLINE,-1,14,17,20,18
     COMBOBOX        C_DEVICE,37,22,226,76,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
-    CONTROL         "@R_PROMISCUS",R_PROMISCUS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,38,37,223,11
+    CONTROL         "@R_PROMISCUOUS",R_PROMISCUOUS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,38,37,223,11
```
Comment (moatazelmasry2) [inline src/PenCore/PenCore.rc]: Fix in Korean

File: src/PenCore/PenCore.rc [hunk 6]
```diff
@@ -4441,7 +4441,7 @@ BEGIN
     LTEXT           "@S_INFO",S_INFO,7,27,293,47
     LTEXT           "@S_TITLE",S_TITLE,7,5,293,21
     CONTROL         "@S_ENABLE",S_ENABLE,"Button",BS_AUTORADIOBUTTON | WS_TABSTOP,48,80,157,12
-    CONTROL         "@S_DISBLE",S_DISBLE,"Button",BS_AUTORADIOBUTTON,48,96,157,12
+    CONTROL         "@S_DISABLE",S_DISABLE,"Button",BS_AUTORADIOBUTTON,48,96,157,12
```
Comment (moatazelmasry2) [inline src/PenCore/PenCore.rc]: Fix in Korean

File: src/PenCore/resource.h [hunk 7]
```diff
@@ -711,7 +711,7 @@
 #define S_TSUKUBA2                      1281
 #define IDC_CHECK1                      1285
 #define R_NO_SAVE_PASSWORD              1285
-#define R_PROMISCUS                     1285
+#define R_PROMISCUOUS                     1285
```
Comment (moatazelmasry2) [inline src/PenCore/resource.h]: Fix indention, so that the number is vertically aligned with other numbers

File: src/bin/hamcore/strtable_cn.stb [hunk 8]
```diff
@@ -21,7 +21,7 @@ DEFAULT_FONT				SimSun
 DEFAULT_FONT_HIGHDPI		Microsoft YaHei
 DEFAULT_FONT_2				SimSun
 DEFAULT_FONT_SIZE			9
-DEFAULE_LOCALE				年 月 日 点 分 秒 星期日 星期一 星期二 星期三 星期四 星期五 星期六 天 小时 分 秒 (无)
+DEFAULT_LOCALE				年 月 日 点 分 秒 星期日 星期一 星期二 星期三 星期四 星期五 星期六 天 小时 分 秒 (无)
```
Comment (moatazelmasry2) [inline src/bin/hamcore/strtable_cn.stb]: Fix in Korean

File: src/bin/hamcore/strtable_cn.stb [hunk 9]
```diff
@@ -1218,7 +1218,7 @@ CM_EDIT_HUB_2				%S 的属性
 
 CM_EDIT_HUB_STANDALONE		当前，服务器正以独立模式运行。而该虚拟 HUB 作为一个独立的 HUB 运行。
 CM_EDIT_HUB_TYPE_FIXED		当前，服务器正以群集模式运行。如果虚拟 HUB 处于这种模式，动态变化将无法进行。
-CM_EDIT_HUB_CREATER			一个新的虚拟 HUB "%S" 已经建立
+CM_EDIT_HUB_CREATED			一个新的虚拟 HUB "%S" 已经建立
```
Comment (moatazelmasry2) [inline src/bin/hamcore/strtable_cn.stb]: Fix in Korean

File: src/bin/hamcore/strtable_cn.stb [hunk 10]
```diff
@@ -5397,7 +5397,7 @@ CMD_CascadePolicySet_VALUE	设定协议的新值，如果协议是数值，请
 CMD_CascadePolicySet_PROMPT_POLNAME	更改值得协议的名称: 
 CMD_CascadePolicySet_PROMPT_POLVALUE	设置新值: 
 CMD_CascadePolicySet_Invalid_Name	指定的协议名称 "%S" 是无效的。\n请运行 PolicyList 命令，您可以获取设置协议的清单。
-CMD_CascadePolicySet_Invalid_Name_For_Cadcade	指定协议 "%S" 在设置级联的安全协议时不可用。
+CMD_CascadePolicySet_Invalid_Name_For_Cascade	指定协议 "%S" 在设置级联的安全协议时不可用。
```
Comment (moatazelmasry2) [inline src/bin/hamcore/strtable_cn.stb]: Fix in Korean

**Comment by jsoref** [comment]
For my records (at some point, I'm going to make this into a script), this is how I performed the Korean bits:
This sets up my histedit:
```sh
hg histedit -r '. % master'
```
^Z (suspend the editor) -- in theory this can be replaced by `EDITOR=`_script_ (that performs the next command)

This picks all the commits that impacted a parallel file and then rewrites the pick lines as edit:
```sh
perl -pi -e 'next unless / '$(hg log -T '{rev}|' -r '. % master'  src/bin/hamcore/strtable_en.stb|sed -e 's/|$//')' /; s/pick/edit/' "/tmp/hg-histedit-tp8NKm.histedit.hg.txt"
```
Return to and quit the default editor:
```
fg
:q
```

This finalizes the current commit; checks to see what the new commit is; and then checks to see what kind of change the current commit is making (is it changing the identifier or changing the translation):
```sh
EDITOR=true hg histedit --cont; hg summary; d -U0 strtable_en.stb|grep $(d -U0 strtable_en.stb |rediff|tail -1|sed -e 's/^-//')
```

This tries to apply the change (including space adjustments-- never relevant for ko, but potentially relevant for other languages; and then a second time w/o space adjustments):
```sh
perl -pi -e "$(d -U0 strtable_en.stb|perl -ne 'next unless /^-[^-]|^\+[^+]/; s/(\s+)\S.*/$1/; chomp; s{^-}{s/}; s{^\+(\S+\s*)}{/$1/;}; print')$(d -U0 strtable_en.stb|perl -ne 'next unless /^-[^-]|^\+[^+]/; s/(\s+)\S.*//; chomp; s{^-}{s/}; s{^\+(\S+\s*)}{/$1/;}; print')" strtable_ko.stb; d -U0 strtable_en.stb strtable_ko.stb
```

File: src/BuildUtil/VpnBuilder.cs [hunk 11]
```diff
@@ -348,7 +348,7 @@ public static class Paths
 		public static readonly string ReleaseDir = Path.Combine(BaseDirName, @"tmp\Release");
 		public static readonly string ReleaseSrckitDir = Path.Combine(BaseDirName, @"tmp\ReleaseSrcKit");
 		public static readonly string StringsDir = Path.Combine(BaseDirName, @"BuildFiles\Strings");
-		public static readonly string CrossCompilerBaseDir = @"S:\CommomDev\xc";
+		public static readonly string CrossCompilerBaseDir = @"S:\CommonDev\xc";
```
[the remark this answers is not in the document]
Comment (jsoref) [reply src/BuildUtil/VpnBuilder.cs]: Those are roughly comments.
```
S:\CommomDev\intel_aes_lib\121209intel_aes\intel-aesni-sample-library-v1.2\Intel_AESNI_Sample_Library_v1.2\intel_aes_lib^@cl^@C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\cl.exe^@cmd^@-FdS:\CommomDev\intel_aes_lib\121209intel_aes\intel-aesni-sample-library-v1.2\Intel_AESNI_Sample_Library_v1.2\intel_aes_lib\vc90.pdb
```

They happen to reference `s:` which is a drive that will not exist on 99% of all computers, so it really isn't a big deal.

If you're concerned, I can certainly move that commit to a distinct PR.

Note: obviously this change will require the user who has an `s:` drive to rename their CommomDev directory to CommonDev. When they do that, they can rebuild the .lib's, and then the .lib's will have the correct spelling.

**Comment by jsoref** [comment]
@moatazelmasry2: Korean changes are included.

File: src/BuildUtil/VpnBuilder.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (jsoref) [reply src/BuildUtil/VpnBuilder.cs]: ok, dropped from this PR. I'm going to just hold off until this lands.

**Comment by chipitsine** [comment]
you can mark it as "WIP:Spelling..." (work in progress) to prevent merging

**Comment by moatazelmasry2** [comment]
Cool thanks. Did you add new commits or just overwrote the last commit. I can see your changes, but I can not see any new commits. Anyways I'll try to get this done today

**Comment by jsoref** [comment]
I rewrote each commit as appropriate. Including excluding the common commit.

**Comment by moatazelmasry2** [comment]
Your great PR has been merged. As a token of gratitude, your name will be added to the list of contributors. Many thanks for your great work

**Comment by domosekai** [comment]
This PR changed spelling in the config file and some RPC packet and broke compatibility with the stable edition.

**Pull request merged**